src/app/shared.module.ts

```diff
@@ -0,0 +1,68 @@
+import { NgModule } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
+import { AppComponent } from './app.component';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+import { ReactiveFormsModule } from '@angular/forms';
+import { TranslateModule } from '@ngx-translate/core';
+import { FormsModule } from '@angular/forms';
+import { HeaderComponent } from './components/header/header.component';
+import { RecipeThumbnailComponent } from './components/recipe-thumbnail/recipe-thumbnail.component';
+import { AutosizeModule } from 'ngx-autosize';
+import { RecipeEditComponent } from './recipe-edit/recipe-edit.component';
+import { MatButtonModule } from '@angular/material/button';
+import { MatInputModule } from '@angular/material/input';
+import { TextFieldModule } from '@angular/cdk/text-field';
+import { MatTableModule } from '@angular/material/table';
+import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
+import { LoginComponent } from './components/login/login.component';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { CommonModule } from '@angular/common';
+import { RouterModule } from '@angular/router';
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    HeaderComponent,
+    RecipeThumbnailComponent,
+    RecipeEditComponent,
+    LoginComponent,
```

Comment:
What about these components? Are there any that should have their own module or should be part of any other module? E.g. RecipeEditComponent might be part of RecipePageModule because I remember it was called from RecipePageComponent.